contrib/test_jenkins.sh

```diff
@@ -306,6 +306,27 @@ get_rdma_device_ip_addr() {
 	echo $ipaddr
 }
 
+get_tcp_ip_addr() {
+    if ! which ibdev2netdev >&/dev/null
+    then
+        return
+    fi
+
+    #get the interface of the ip address that is used to reach The Google Public DNS IP addresses (IPv4).
+    #making the assumption that the pure Ethernet IPv4 address on the host is the one used for this purpose.
+    eth_iface=$(ip route get 8.8.8.8 | awk -F 'dev' '/dev/{print $2}' | awk '{print $1}')
+
+    #the pure Ethernet interface should not appear in the ibdev2netdev output. it should not be an IPoIB or
+    #RoCE interface.
+    if [ -z "$(ibdev2netdev | grep -w ${eth_iface})" ] ; then
+      ipaddr=$(ip addr show "${eth_iface}" | awk '/inet /{print $2}' | awk -F '/' '{print $1}')
+    else
+      echo "Failed to retrieve an IP of a non IPoIB/RoCE interface" >&2
```

Comment:
echo ...
exit 1
or
echo ...
false

Reply:
@yosefe can you please clarify? i'm not sure what you mean to change here

Reply:
don't use stderr redirection to make script exit

Reply:
i see.
i didn't want to kill Jenkins if there is no eth IP to test, but ok, will change this.

Reply:
ok, then no need to redirect to stderr (and also no need to exit 1), just skip this test

Reply:
this is what happens now if there is no ip - the error will be printed, nothing would be returned and the fallback won't be tested.

but now that you mentioned it, maybe we do need to force testing an eth ip? if we don't, we might end up not testing the fallback in jenkins for a long while due to a misconfiguration or something on the hosts.

Reply:
> but now that you mentioned it, maybe we do need to force testing an eth ip? if we don't, we might end up not testing the fallback in jenkins for a long while due to a misconfiguration or something on the hosts.

ok, we should expect to always have an Eth device for TCP